extra/modules/greenbids-real-time-data/src/main/java/org/prebid/server/hooks/modules/greenbids/real/time/data/model/predictor/OnnxModelRunnerWithThresholds.java

```diff
@@ -0,0 +1,63 @@
+package org.prebid.server.hooks.modules.greenbids.real.time.data.model.predictor;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.github.benmanes.caffeine.cache.Cache;
+import com.google.cloud.storage.Storage;
+import io.vertx.core.Future;
+import io.vertx.core.Vertx;
+import org.prebid.server.hooks.modules.greenbids.real.time.data.core.Partner;
+import org.prebid.server.hooks.modules.greenbids.real.time.data.core.ThrottlingThresholds;
+
+public class OnnxModelRunnerWithThresholds {
+
+    private final Cache<String, OnnxModelRunner> modelCacheWithExpiration;
+    private final Cache<String, ThrottlingThresholds> thresholdsCacheWithExpiration;
+    private final String gcsBucketName;
+    private final String onnxModelCacheKeyPrefix;
+    private final String thresholdsCacheKeyPrefix;
+    private final Storage storage;
+    private final Vertx vertx;
+
+    public OnnxModelRunnerWithThresholds(
+            Cache<String, OnnxModelRunner> modelCacheWithExpiration,
+            Cache<String, ThrottlingThresholds> thresholdsCacheWithExpiration,
+            Storage storage,
+            String gcsBucketName,
+            String onnxModelCacheKeyPrefix,
+            String thresholdsCacheKeyPrefix,
+            Vertx vertx) {
+        this.modelCacheWithExpiration = modelCacheWithExpiration;
+        this.thresholdsCacheWithExpiration = thresholdsCacheWithExpiration;
+        this.gcsBucketName = gcsBucketName;
+        this.onnxModelCacheKeyPrefix = onnxModelCacheKeyPrefix;
+        this.thresholdsCacheKeyPrefix = thresholdsCacheKeyPrefix;
+        this.storage = storage;
+        this.vertx = vertx;
+    }
+
+    public Future<OnnxModelRunner> retrieveOnnxModelRunner(Partner partner) {
+        final String onnxModelPath = "models_pbuid=" + partner.getPbuid() + ".onnx";
+        final ModelCache modelCache = new ModelCache(
+                onnxModelPath,
+                storage,
+                gcsBucketName,
+                modelCacheWithExpiration,
+                onnxModelCacheKeyPrefix,
+                vertx);
```

Comment:
create each time a new ModelCache is something counter intuitive 

create ModelCache bean with the `modelCache.get(onnxModelPath, partner.getPbuid())`

Reply:
defined `modelCache` and `ThresholdCache` as beans + provided paths at  and `get` methods